swaggen/src/main/java/generator/PathGenerator.java

```diff
@@ -54,9 +65,86 @@
+		for (PathInfo existingPath: existingPaths.keySet()) {
+			for (PathInfo pathKey : classPaths.keySet()) {
+				if ((existingPath.getBasePath().equals(pathKey.getBasePath())) && 
+					(existingPath.getURI().equals(pathKey.getURI()))) {
+					existingPaths.get(existingPath).addEndpoint(classPaths.get(pathKey));
+					classPaths.remove(pathKey);
+				}
+			}
+		}
+		existingPaths.putAll(classPaths);
+		return existingPaths;
+	}
+
+	/**
+	 * Checks if there are any other request methods within a class
+	 * and puts it in the swagger endpoints map
+	 * Case: multiple request methods within the same class
+	 * @param annotation SwaggerGen annotation
+	 * @param existingPaths Map of existing Swagger Endpoints
+	 */
+	private static void checkRequestMethods(SwaggerGen annotation, Map<PathInfo, SwaggerEndpoint> existingPaths, SwaggerGenClass klassAnnotation) {
+		if (annotation == null) {
+			throw new IllegalArgumentException("annotation cannot be null");
+		}
+		PathInfo pathInfo = addPathInfo(annotation, klassAnnotation);
+
+		if (existingPaths.isEmpty()) {
+			existingPaths.put(pathInfo, generatePathFromAnnotation(annotation, klassAnnotation));
+		}
+		for (PathInfo existingPath: existingPaths.keySet()) {
+			if ((existingPath.getBasePath().equals(pathInfo.getBasePath())) && 
+			(existingPath.getURI().equals(pathInfo.getURI()))) {
+				existingPaths.get(existingPath).addEndpoint(generatePathFromAnnotation(annotation, klassAnnotation));
+			} else {
+				existingPaths.put(pathInfo, generatePathFromAnnotation(annotation, klassAnnotation));
+			}
+		}
+	}
+
+	/**
+	 * Creates a new PathInfo object that holds the server url, path base path and uri
+	 * @param annotation method level annotation information
+	 * @param klassAnnotation class level annotation information
+	 * @return PathInfo object with necessary information
+	 */
+	private static PathInfo addPathInfo(SwaggerGen annotation, SwaggerGenClass klassAnnotation) {
+		String basePath = (annotation.basePath().isBlank() && klassAnnotation != null) ? klassAnnotation.basePath() : annotation.basePath();
+		String[] servers = (klassAnnotation != null) ? klassAnnotation.servers() : new String[0];
+		ArrayList<ServerURL> serverURL = new ArrayList<ServerURL>(); 
+
+		for (String server: servers) {
+			try {
+				String[] urlAndDesc = server.split("=", 2);
+				String serverUrl = urlAndDesc[0].replaceAll("\\s+", "");
+				String serverDescription = urlAndDesc[1].trim();
+				
+				HashMap<String, ServerVariable> serverVariables = ServerVariable.addServerVariables(serverUrl, klassAnnotation);
+
+				ServerURL newServerUrl = new ServerURL(serverUrl, serverDescription, serverVariables); 
+				serverURL.add(newServerUrl);
+
+			} catch (ArrayIndexOutOfBoundsException ex) { 
+				throw new ArrayIndexOutOfBoundsException("Declared server incorrectly, missing \"=\" sign with class annotation server declaration.");

```

Comment:
There are 2 issues here:

1. Make a custom exception for this: `InvalidServerDefinitionException`. Or throw an `IllegalArgumnetException`. This comes with experience, but in general: we don't want to throw exceptions that indicate a problem with the code.

For example:
Don't throw a `NullPointerException` because it indicates that we (as programmers) have tried to access a property of a null value. We were missing appropriate null checks or forgot to set something.
Don't throw an `ArrayIndexOutOfBoundsException` as it indicates that we (as programmers) didn't check the size of the array before arbitrarily accessing it.

2. We should preemptively validate the server definition. A future programmer may come along and not understand why an `ArrayIndexOutOfBoundsException` means that the server was improperly defined. I recommend creating a new function:
```
// Rough Pseudocode
parsePropertyWithDescription(stringToParse, seperator) throws ParsingException
    parsingRegex <- // Define a regex to split the String on the separator. (If you need help with regex, let me know)
    property <- parsingRegex.property
    description <- parsingRegex.description
    return new Pair<String, String>(property, dexcription) // Can be a pair or a custom class.
```

then you can catch the `ParsingException` and throw a more appropriate one (as discussed in issue 1). This method would be useful elsewhere in the code, so it would be very beneficial.